core/commands/files/files.go

```diff
@@ -0,0 +1,555 @@
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		src := req.Arguments()[0]
+		dst := req.Arguments()[1]
+
+		err = mfs.Mv(n.FilesRoot, src, dst)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+	},
+}
+
+var FilesWriteCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Write to a mutable file in a given filesystem",
+		ShortDescription: `
+Write data to a file in a given filesystem. This command allows you to specify
+a beginning offset to write to. The entire length of the input will be written.
+
+If the '--create' option is specified, the file will be create if it does not
+exist. Nonexistant intermediate directories will not be created.
+
+Example:
+
+	echo "hello world" | ipfs files write --create /myfs/a/b/file
+	echo "hello world" | ipfs files write --truncate /myfs/a/b/file
+		`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("path", true, false, "path to write to"),
+		cmds.FileArg("data", true, false, "data to write").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.IntOption("o", "offset", "offset to write to"),
+		cmds.BoolOption("n", "create", "create the file if it does not exist"),
+		cmds.BoolOption("t", "truncate", "truncate the file before writing"),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		path := req.Arguments()[0]
+		create, _, _ := req.Option("create").Bool()
+		trunc, _, _ := req.Option("truncate").Bool()
+
+		nd, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		fi, err := getFileHandle(nd.FilesRoot, path, create)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		defer fi.Close()
+
+		if trunc {
+			if err := fi.Truncate(0); err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+		}
+
+		offset, _, _ := req.Option("offset").Int()
+
+		_, err = fi.Seek(int64(offset), os.SEEK_SET)
+		if err != nil {
+			log.Error("seekfail: ", err)
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		input, err := req.Files().NextFile()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		n, err := io.Copy(fi, input)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		log.Debugf("wrote %d bytes to %s", n, path)
+	},
+}
+
+var FilesMkdirCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "make directories",
+		ShortDescription: `
+Create the directory if it does not already exist.
+
+Note: all paths must be absolute.
+
+Examples:
+
+    $ ipfs mfs mkdir /test/newdir
+	$ ipfs mfs mkdir -p /test/does/not/exist/yet
+`,
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("path", true, false, "path to dir to make"),
+	},
+	Options: []cmds.Option{
+		cmds.BoolOption("p", "parents", "no error if existing, make parent directories as needed"),
```

Comment:
- "no error if existing" ? 
- instead: "create intermediary directories as required"

should we be able to pass a template like to `ipfs object patch` ? or does `mfs` assume everything is unixfs files? (it may be very valuable to drop that assumption in the future)